src/hooks/useConnectionStatus.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useContext, useEffect, useState } from 'react'
+
+export type ConnectionStatus = 'connected' | 'disconnected' | 'connecting'
+
+const ConnectionStatusContext = React.createContext<undefined | ConnectionStatusValue>(undefined)
+ConnectionStatusContext.displayName = 'ConnectionStatusContext'
+
+type ConnectionStatusValue = {
+  setNodeConnection: (connection: ConnectionStatus) => void
+  nodeConnectionStatus: ConnectionStatus
+  isUserConnectedToInternet: boolean
+}
+
+const withTimeout = async <T,>(promise: Promise<T>, timeout: number) => {
+  const timeoutPromise = new Promise<T>((resolve, reject) => setTimeout(() => reject(new Error('Timed out!')), timeout))
+  return await Promise.race([timeoutPromise, promise])
+}
+
+export const ConnectionStatusProvider: React.FC = ({ children }) => {
+  const [nodeConnectionStatus, setNodeConnection] = useState<ConnectionStatus>('connecting')
+  const [isUserConnectedToInternet, setIsUserConnectedToInternet] = useState(true)
+
+  useEffect(() => {
+    // ping google every three seconds to check if user is connected to internet
+    const interval = setInterval(() => {
+      checkConnection()
+    }, 3000)
+    return () => {
+      clearInterval(interval)
+    }
+  }, [])
```

Comment:
It's amazing that ESLint doesn't complain about `checkConnection` not missing from here. I think it can tell that `checkConnection` doesn't contain any dynamic references so it's safe to just use the stale reference. Magic